upgrade/upgrade_provider.go

```diff
@@ -298,3 +306,23 @@ func tfgenAndBuildSDKs(
 		InformGitHub(ctx, upgradeTarget, repo, goMod, targetBridgeVersion, tfSDKUpgrade, os.Args)
 	}
 }
+
+func BumpRecordedUpstreamVersion(ctx context.Context, version *semver.Version, configFile string) error {
+	return stepv2.PipelineCtx(ctx, "Bump Recorded Upstream Version", func(ctx context.Context) {
+		content := stepv2.ReadFile(ctx, configFile)
+		lines := strings.Split(content, "\n")
+		var newLines []string
+		for _, line := range lines {
+			if !strings.Contains(line, "current-upstream-version:") {
+				newLines = append(newLines, line)
+			}
+		}
+		content = strings.Join(newLines, "\n")
+		content += "current-upstream-version: " + version.String() + "\n"
```

Comment:
Is not this file in YAML? Can we write a struct for it, document the fields and read/write using YAML tooling instead of munging text?